### ISSUE TITLE
bug(false-positive): quick-lint does not allow a type and variable to have the same name

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#981]
steps to reproduce:

```ts
let a = 1
type a = undefined
```

expected:
lints fine

actual:
```
you couldn't get enough of a, so you had to make two, huh?
```

While this syntax is confusing, it's frequently used to use `const` object literals as enums.  Like

```ts
export const Thing = {
  "A": 1,
  "B": 2
} as const;

type Thing = typeof Thing
```

**quick-lint-js version 2.12.0**

**Comment by leszek888** [comment]
I could take this one.

**Comment by strager** [comment]
@leszek888 Sorry, but I am working on this myself.

**Comment by strager** [comment]
This has been fixed. I don't know when exactly, but it's working on master and probably quick-lint-js 2.15.0.
